plot/plot.app.js

```diff
@@ -683,14 +695,56 @@
                                                     }
 
                                                     data.forEach(function (row) {
+                                                        $rootScope.templateParams.$self = {
+                                                            data: row
+                                                        }
                                                         if(label) {
-                                                            values.labels.push(row[trace.legend_col]);
+                                                            var traceLabel=row[trace.legend_col];
+                                                            //Changes related to adding markdown templating pattern to legends in pie
+                                                            if(trace.hasOwnProperty("legend_markdown_pattern") && trace.legend_markdown_pattern){
+                                                                // Check if the ? is present in the markdown pattern if yes add & else add ?
+                                                                var qCharacter = /\((.*?)\)/ig.exec(trace.legend_markdown_pattern)[1] !== -1 ? "&" : "?";
+                                                                try{
+                                                                    traceLabel=ERMrest.renderMarkdown(ERMrest.renderHandlebarsTemplate(trace.legend_markdown_pattern,$rootScope.templateParams),true);
+                                                                    //Defined regex pattern to extract the url in markdown pattern
+                                                                    //Example : traceLabel: <a href="/chaise/recordset/#2/Gene_Expression:Specimen/*::facets::N4IghgdgJiBcDaoDOB7ArgJwMYFM6JHQBcAjdafEAQQiIEsyoBPEAGmtoZWYH0AVHEiJIeAYRQRUGemgC2PAGYBrHCwC6AX3YAlAJIAREGvZYAFijq4klAIoBaAMwBpAKwA2I1uTpseBKGIyNAoEDnpGFnYacO4mfkFhMUwwegkeAGUiFLQRZVVPHQMjE3NLQVs7AE03AEYAWU9NIA&pcid=plotApp&ppid=2oio1qej2i6m2bv62ox21h1q" target="_blank">Imaging: Histology</a>
+                                                                    // extractedLink = /chaise/recordset/#2/Gene_Expression:Specimen/*::facets::N4IghgdgJiBcDaoDOB7ArgJwMYFM6JHQBcAjdafEAQQiIEsyoBPEAGmtoZWYH0AVHEiJIeAYRQRUGemgC2PAGYBrHCwC6AX3YAlAJIAREGvZYAFijq4klAIoBaAMwBpAKwA2I1uTpseBKGIyNAoEDnpGFnYacO4mfkFhMUwwegkeAGUiFLQRZVVPHQMjE3NLQVs7AE03AEYAWU9NIA&pcid=plotApp&ppid=2oio1qej2i6m2bv62ox21h1q
+                                                                    var extractedLink=/<a\b.+href="([^\n\r]*)"\s/ig.exec(traceLabel)[1];
+                                                                    var linkWithContextParams =extractedLink+qCharacter+"pcid="+contextUrlParams.cid+"&ppid="+contextUrlParams.pid;
+                                                                    values.legend_markdown_pattern.push(linkWithContextParams);
```

Comment:
Is this the link that is being used for the click event when a legend value is clicked?

If so, we should give it a more appropriate name since it won't be a markdown pattern anymore (we already ran renderMarkdown above).

Reply:
okay.. we can change the name